tfx/experimental/pipeline_testing/dummy_executor.py

```diff
@@ -0,0 +1,71 @@
+# Lint as: python2, python3
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Base Dummy Executor"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+from typing import Any, Dict, List, Text, Optional
+
+import absl
+
+from tfx import types
+from tfx.components.base import base_executor
+from tfx.utils import io_utils
+
+class BaseDummyExecutor(base_executor.BaseExecutor):
```

Comment:
I'm curious about `BaseDummyExecutor` is the proper name for the behavior of this class. Because both of `base` and `dummy` are quite abstract terms, and this class is not designed to be extended. It would be better to give a more concrete name, for example, `ReplayingExecutor`, `StubExecutor` or `FixedResultExecutor`. (And the name of the file should follow the class name.)

And I think that it would be better to have more comprehensive description in docstring.

@nikelite Do you have any idea on this?

Reply:
Users would extend this class, if they want to put their custom dummy executors, but I agree that we could improve the name.